editor/src/clj/editor/model.clj

```diff
@@ -137,43 +166,254 @@
+            (g/set-property binding :textures textures)
+            (g/connect binding :_node-id self :nodes)
+            (g/connect binding :dep-build-targets self :dep-build-targets)
+            (g/connect binding :scene-info self :scene-infos)
+            (g/connect binding :material-binding-info self :material-binding-infos)))
+        (for [[old-material-binding-name+order [_ {:keys [material textures]}]] changed
+              :let [binding (:_node-id (old-material-binding-name+order->info old-material-binding-name+order))]
+              tx [(g/set-property binding :material material)
+                  (g/set-property binding :textures textures)]]
+          tx)
+        (for [[old-material-binding-name+order [new-material-binding-name]] renamed]
+          (g/set-property (:_node-id (old-material-binding-name+order->info old-material-binding-name+order))
+            :name new-material-binding-name))
+        (for [[material-binding-name+order] removed]
+          (g/delete-node (:_node-id (old-material-binding-name+order->info material-binding-name+order))))))))
+
+(defn- add-material [model-node-id material-name material-resource]
+  (g/update-property model-node-id :materials conj {:name material-name
+                                                    :material material-resource
+                                                    :textures []}))
+
+(defn- remove-material [model-node-id material-name]
+  (g/update-property model-node-id
+                     :materials
+                     (fn [materials]
+                       (filterv #(not= material-name (:name %)) materials))))
+
+(def ^:private fake-resource
+  (reify resource/Resource
+    (children [_])
+    (ext [_] "")
+    (resource-type [_])
+    (source-type [_])
+    (exists? [_] false)
+    (read-only? [_] true)
+    (path [_] "")
+    (abs-path [_] "")
+    (proj-path [_] "")
+    (resource-name [_] "")
+    (workspace [_])
+    (resource-hash [_])
+    (openable? [_] false)
+    (editable? [_] false)))
+
+(g/defnk produce-model-properties [_node-id _declared-properties material-binding-infos mesh-material-ids]
+  (let [model-node-id _node-id
+        mesh-id? (set mesh-material-ids)
+        proto-ids (into #{} (map :name) material-binding-infos)
+        new-props (-> []
+                      ;; existing materials / textures:
+                      (into
+                        (mapcat
+                          (fn [{:keys [_node-id material name texture-binding-infos]}]
+                            (let [should-be-deleted (not (mesh-id? name))]
+                              (into [[(keyword (str "__material__" name))
+                                      (cond-> {:node-id _node-id
+                                               :label name
+                                               :type resource/Resource
+                                               :value (cond-> material should-be-deleted (or fake-resource))
```

Comment:
We implement removing the material binding using the clear button (like clear override). This requires a workaround though: the clear button is only shown when the value is non-nil, but we need to show it even when it's nil, so we use a fake resource to pretend there is a value.